image_provisioner/playbooks/roles/cri-o/tasks/main.yml

```diff
@@ -0,0 +1,98 @@
+---
+- block:
+    - name: Ensure container-selinux is installed
+      package:
+        name: container-selinux
+        state: present
+
+    # Used to pull and install the system container
+    - name: Ensure atomic is installed
+      package:
+        name: atomic
+        state: present
+
+    - name: Ensure runc is installed
+      package:
+        name: runc
+        state: present
+  when: not atomic | default(False) | bool
+
+- name: Use RHEL based image
+  set_fact:
+    crio_image_prepend: "{{ crio_registry }}"
+    crio_image_name: "cri-o"
+
+- name: Set the full image name
+  set_fact:
+    crio_image: "{{ crio_image_prepend }}/{{ crio_image_name }}:latest"
+
+- name: Pre-pull CRI-O System Container image
+  command: "atomic pull --storage ostree {{ crio_image }}"
+
+- name: Install CRI-O container
+  command: "atomic install --system-package no --system {{ crio_image }}"
+
+- name: Ensure that selinux is set to true in the config
+  lineinfile:
+    path: /etc/crio/crio.conf
+    regexp: '^selinux'
+    line: 'selinux = true'
+
+- name: Ensure that systemd is the cgroup manager
+  lineinfile:
+    path: /etc/crio/crio.conf
+    regexp: '^cgroup_manager'
+    line: "cgroup_manager = \"systemd\""
+
+- name: Ensure that file locking is set to false
+  lineinfile:
+    path: /etc/crio/crio.conf
+    regexp: '^file_locking'
+    line: 'file_locking = false'
+
+- name: override overlay2 kernel check
+  lineinfile:
+    path: /etc/crio/crio.conf
+    insertafter: '^storage_option'
+    line: "\t\"overlay2.override_kernel_check=1\""
+    state: present
+
+- name: create cni directory to store network settings
+  file:
+    path: /etc/cni/net.d
+    state: directory
+
+- name: create directory to store plugins
+  file:
```

Comment:
As above, do we have two create directory steps due to how it was done upstream?

Reply:
Yes.